### PR TITLE
I1 Loan

### PR DESCRIPTION
**Pull request by elliotmitchum**
Refactors the `Loan` class.

Criteria:

- Evaluate against the class diagram
- Evaluate against the code guidelines
- Project builds and runs

### CODE REVIEW

**Comment by elliotmitchum** [comment]
@amrmosit This is ready for review.

**Comment by amrmosit** [comment]
Thanks @elliotmitchum 
Hey @haynescrossley, this PR is ready for review.
Code is operational (Checked)

File: src/library/entities/Loan.java [hunk 1]
```diff
@@ -1,80 +1,83 @@
...
+
+    private enum LoanState {CURRENT, OVER_DUE, DISCHARGED}
+
+    private long loanId;
+
+    private Item item;
+
+    private Patron patron;
+
+    private Date date;
```
Comment (haynescrossley) [inline src/library/entities/Loan.java]: [Class diagram](https://interact2.csu.edu.au/webapps/blackboard/execute/content/file?cmd=view&content_id=_5279818_1&course_id=_63244_1) has this Date variable listed as dueDate. Please refactor/rename.

File: src/library/entities/Loan.java [hunk 2]
```diff
@@ -1,80 +1,83 @@
...
+    }
+
+
+    public Long getId() {
+        return loanId;
+    }
+
+
+    public Date getDueDate() {
+        return date;
```
Comment (haynescrossley) [inline src/library/entities/Loan.java]: I assume this should return dueDate instead of date. This should be fixed with the refactor/rename above.

**Review by haynescrossley — CHANGES_REQUESTED** [review]
Only one class variable name to refactor/rename in line with class diagram. Otherwise good to go.
Cheers,
Haynes

**Review by haynescrossley — APPROVED** [review]
Changes approved. Ready for merge @amrmosit.

Changes addressed via commit:

Rename date to dueDate
e724d70

**Comment by amrmosit** [comment]
Loan has been merged into master

**Pull request merged**